sphinxlint.py

```diff
@@ -255,11 +268,62 @@ def paragraphs(lines):
         yield paragraph_lno, "".join(paragraph)
 
 
+QUOTE_PAIRS = [
+    '»»',  # Swedish
+    '‘‚',  # Albanian/Greek/Turkish
+    '’’',  # Swedish
+    '‚‘',  # German
+    '‚’',  # Polish
+    '“„',  # Albanian/Greek/Turkish
+    '„“',  # German
+    '„”',  # Polish
+    '””',  # Swedish
+    '››',  # Swedish
+    "''",  # ASCII
+    '""',  # ASCII
+    "<>",  # ASCII
+    "()",  # ASCII
+    "[]",  # ASCII
+    "{}",  # ASCII
+]
+
+QUOTE_PAIRS_NEGATIVE_LOOKBEHIND = "(?<!" + (
+    "|".join(f"{re.escape(left)}`{re.escape(right)}" for left, right in QUOTE_PAIRS) +
+    "|" +
+    "|".join(f"{re.escape(openers[i])}`{re.escape(closers[i])}" for i in range(len(openers)))
+) + ")"
+
 role_body = rf"([^`]|\s`+|\\`|:{simplename}:`([^`]|\s`+|\\`)+`)+"
 normal_role = f":{simplename}:`{role_body}`"
 backtick_in_front_of_role = re.compile(rf"(^|\s)`:{simplename}:`{role_body}`")
 
 
+# https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-markup-recognition-rules
+default_role_re = re.compile(
+    r"""
+    (?<!\x00) # Both, inline markup start-string and end-string must not be preceded by an unescaped backslash
+    (?<=      # Inline markup start-strings must:
+        ^|           # start a text block
+        \s|          # or be immediately preceded by whitespace,
+        [-:/'"<([{]| # one of the ASCII characters
+        [\p{Ps}\p{Pi}\p{Pf}\p{Pd}\p{Po}]  # or a similar non-ASCII punctuation character.
+    )
+    (?P<default_role>`        # Default role start
+    [^`\s]    # Inline markup start-strings must be immediately followed by non-whitespace.
+              # The inline markup end-string must be separated by at least one character from the start-string.
+    """ + QUOTE_PAIRS_NEGATIVE_LOOKBEHIND + r"""
+    [^`]*
+    `)        # Default role end
```

Comment:
Should this be indented?
```suggestion
    (?P<default_role>`        # Default role start
        [^`\s]    # Inline markup start-strings must be immediately followed by non-whitespace.
                  # The inline markup end-string must be separated by at least one character from the start-string.
        """ + QUOTE_PAIRS_NEGATIVE_LOOKBEHIND + r"""
        [^`]*
    `)        # Default role end
```